toolchain/base/value_store.h

```diff
@@ -81,7 +81,8 @@ class ValueStore {
  public:
   // Stores the value and returns an ID to reference it.
   auto Add(typename IdT::IndexedType value) -> IdT {
-    auto id = IdT(values_.size());
+    IdT id = IdT(values_.size());
+    CARBON_CHECK(id.index >= 0) << "Too many numeric literals";
```

Comment:
Please fix the error message to be generic.

Reply:
Ah, I see this is used for more types in later PRs.